src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -283,6 +283,12 @@ private EventHubAdapterReceiver GetOrCreateReceiver(QueueId queueId)
             return cache;
         }
 
+        protected virtual IEventHubQueueCache CreateCache(IStreamQueueCheckpointer<string> checkpointer, 
```

Comment:
The pattern for customizing the cache is to provide one's own CacheFactory.  That's why we check to see if it is null before assigning it in the initialization, so a specialization can provide it's own.  Having this virtual function as an alternate incomplete extension point, imo, confuses the pattern.  I concede that assigning a CacheFactory is probably not the best approach, but I don't think we should have multiple extensibility patterns.